### DISCUSSION TITLE
Brew version of youtube-dl seems out-of-date

### DISCUSSION THREAD

**Discussion by shlomiv** [Homebrew/discussions#4329] [Everyday usage] [answered]
### Output of `brew config`

```shell
brew config                                                                                                                                                                                             ~
HOMEBREW_VERSION: 3.5.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: c5731faf8be1ddc1eeae4b162b88bd76318fc279
Last commit: 7 months ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0398c0937e7f745c93e3daf21f9a82ddcd17e5a6
Core tap last commit: 7 months ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8_1/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0.5 build 1205
Git: 2.34.1 => /opt/homebrew/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.4-arm64
CLT: 12.5.0.22.11
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Preferences or run:
  softwareupdate --all --install --force

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 13.2.1.


Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  szip

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Library/Frameworks/Python.framework/Versions/3.11/bin/python3-config
  /Library/Frameworks/Python.framework/Versions/3.11/bin/python3.11-config

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  six
  yamllint

Warning: /usr/bin occurs before /opt/homebrew/bin in your PATH.
This means that system-provided programs will be used instead of those
provided by Homebrew. Consider setting your PATH so that
/opt/homebrew/bin occurs before /usr/bin. Here is a one-liner:
  echo 'export PATH="/opt/homebrew/bin:$PATH"' >> ~/.zshrc

The following tools exist at both paths:
  git-receive-pack
  python3
  pip3
  git-upload-archive
  git-shell
  git-upload-pack
  git

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /opt/homebrew/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/opt/homebrew/sbin:$PATH"' >> ~/.zshrc
```

### Description of issue

Not sure where to ask for this, so hope I am in the right spot. 

I asked [youtube-dl maintainers](https://github.com/ytdl-org/youtube-dl/issues/31530#issuecomment-1470954253) about when one of their fixes would be brewable, and they turned me here, to ask you for the following:

>  ask them below to package youtube-dl from a recent master branch code snapshot instead of from last (but now deprecated) tagged release (as a new one isn't imminent - not yet, at least...):
>
> https://github.com/Homebrew/discussions/discussions

Would that be something that you guys do? I have no clue how this works, just want the package to work as expected.

Thanks!

**Comment by gromgit** [comment]
Not a Homebrew maintainer, but "recent master branch code snapshot" is entirely arbitrary, and easily leads to _daily_ builds (or worse) as users clamor for the latest "must-have" feature that was just committed an hour ago.

Also, the `yt-dlp` folks don't seem to have the same issue with regular releases, so I've long since switched to them.

**Reply by shlomiv** [reply]
Thanks for the reply @gromgit! 

Yes, basing on `master` seems a bit risky to me as you mentioned.. Not sure why its that way, maybe @Vangelis66 would be able to shed some light about this decision.

In any event, I was able to solve this outside of brew, with the following snippet (uninstall brew version of youtube-dl first):
```
sudo curl -L https://github.com/ytdl-patched/youtube-dl/releases/latest/download/youtube-dl -o /usr/local/bin/youtube-dl
sudo chmod a+rx /usr/local/bin/youtube-dl
```

This seems to install `latest` version directly.

Thanks for the tip regarding [yt-dlp](https://github.com/yt-dlp/yt-dlp), I may give it a try!

**Reply by Vangelis66** [reply]
... As I posted in yt-dl's tracker, I'm not affiliated with the current `youtube-dl` maintainer, simply volunteering there 😜 ...

> maybe @Vangelis66 would be able to shed some light about this decision.

"New release" discussion can be found below: 

https://github.com/ytdl-org/youtube-dl/issues/31585 (several duplicates of that issue exist); pending **PR** is: 

https://github.com/ytdl-org/youtube-dl/pull/30644

At this point, the best approach is to "update" via installing from latest git source (e.g. pip git+https...); or use the binaries kindly offered at: 
https://github.com/ytdl-patched/youtube-dl/

> the `yt-dlp` folks don't seem to have the same issue with regular releases

... Well, _they_ do **monthly stable** releases (but nightly ones, too), so, perhaps, the **Homebrew** team could cut and package "monthly" releases from `youtube-dl`'s `git-master`, until the "stable release" problems have been sorted... Just my 2c...

**Reply by shlomiv** [reply]
Thank you @Vangelis66, for referencing the monthly builds, hopefully this is something brew could use!

Also thank you for volunteering on `youtube-dl`, you sure got me fooled as a maintainer 😅

**Reply by gromgit** [reply]
> > the `yt-dlp` folks don't seem to have the same issue with regular releases
> 
> ... Well, _they_ do **monthly stable** releases (but nightly ones, too), so, perhaps, the **Homebrew** team could cut and package "monthly" releases from `youtube-dl`'s `git-master`, until the "stable release" problems have been sorted...

The "they" is key here. Upstream tags are an implicit promise that "this is a stable and easily identifiable version that should be safe to distribute, report issues on, and revert to in the worst case". It would be foolish for large-scale downstream distributors like Homebrew to choose a commit on one project, with zero knowledge about the underlying codebase (to say nothing about any pending deal-breaking issues on the same), and say "OK, even though upstream doesn't think they're ready to commit to a release, we'll make that decision for them". It would be positively _insane_ for them to do so across the _thousands_ of projects they distribute binaries for.

As for https://github.com/ytdl-patched/youtube-dl/, it's clearly marked "unofficial". Given the Homebrew maintainers' track record of refusing to update stale projects to a more lively fork without a clear "blessing" from the original project owners, I doubt they'll want to use this project as a "source of truth" for stable releases.

**Comment by carlocab** [comment, marked as answer]
I've opened Homebrew/homebrew-core#125804. When that is merged, you can use
```
brew install --HEAD youtube-dl
```
to install `youtube-dl` from the tip of the `master` branch. Note that this install will no longer be automatically updated by `brew upgrade`, and that we don't provide support for `--HEAD` installs (because they require building from source and these can fail for any reason). However, if you're feeling adventurous you may wish to give it a try.

**Reply by shrinidhi111** [reply]
I am unable to perform this step, as brew has stopped supporting youtube-dl

```
MacBook-Pro-2:youtube si$ brew install --HEAD youtube-dl
Error: youtube-dl 2021.12.17 is already installed
To install HEAD, first run:
  brew unlink youtube-dl
MacBook-Pro-2:youtube si$ brew uninstall youtube-dl
Uninstalling /opt/homebrew/Cellar/youtube-dl/2021.12.17... (1,661 files, 11.5MB)
MacBook-Pro-2:youtube si$ 
MacBook-Pro-2:youtube si$ brew install --HEAD youtube-dl
Error: youtube-dl has been disabled because it has a failing test since forever and no new release since 2021! It was disabled on 2024-11-23.
```

**Reply by gromgit** [reply]
[yt-dlp](https://github.com/yt-dlp/yt-dlp) is its spiritual and practical replacement.
```
brew rm youtube-dl
brew install yt-dlp
man yt-dlp
```

**Reply by shrinidhi111** [reply]
> [yt-dlp](https://github.com/yt-dlp/yt-dlp) is its spiritual and practical replacement.
> 
> ```
> brew rm youtube-dl
> brew install yt-dlp
> man yt-dlp
> ```


Thank you!

Its been many years since I was actively downloading youtube videos. This helps

**Comment by exapsy** [comment]
Hey folks, it appears that the message now when you try to install youtube-dl is this

```sh
$ brew install --HEAD youtube-dl

Error: youtube-dl has been disabled because it has a failing test since forever and no new release since 2021! It was disabled on 2024-11-23.
```

Someone even seems to have got passive aggressive there 😂

**Comment by shrinidhi111** [comment]
Cannot install youtube-dl using homebrew anymore


```
MacBook-Pro-2:youtube si$ brew install --HEAD youtube-dl
Error: youtube-dl has been disabled because it has a failing test since forever and no new release since 2021! It was disabled on 2024-11-23.

```

**Reply by gromgit** [reply]
https://github.com/orgs/Homebrew/discussions/4329#discussioncomment-11888191
